### PR TITLE
Add Intel hardware intrinsic API implementation to mscorlib

### PR DESCRIPTION
**Pull request by fiigii**
This PR is the complete design of [API Proposal: Add Intel hardware intrinsic functions and namespace #22940](https://github.com/dotnet/corefx/issues/22940) and the mscorlib counterpart of [Add Intel hardware intrinsic APIs to CoreFX #23489](https://github.com/dotnet/corefx/pull/23489).

For building with the current code base, `const` parameter modifiers and `[intrinsic]` is temporarily removed.

### CODE REVIEW

**Comment by fiigii** [comment]
cc @russellhadley @terrajobst @CarolEidt @mellinoe

File: src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the http://aka.ms/api-review process.
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs]: This comment does not apply here.

Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs]: I see. Will fix.

File: src/mscorlib/src/System/Runtime/Intrinsics/X86/Aes.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+using System;
+using System.Runtime.Intrinsics;
+
+namespace System.Runtime.Intrinsics.X86
+{
+    /// <summary>
+    /// This class provides access to Intel AES hardware instructions via intrinsics
+    /// </summary>
+
```
Comment (justinvp) [inline src/mscorlib/src/System/Runtime/Intrinsics/X86/Aes.cs]: Nit: Unnecessary blank line

Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/X86/Aes.cs]: Will remove.

File: src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics
+{
+    [StructLayout(LayoutKind.Sequential, Size = 32)]
+    public struct Vector256<T> where T : struct 
+    {
+        public static bool IsSupported {get;}
+    }
+}
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs]: Nit: Add new lines at the end of the file

File: src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics
+{
+    [StructLayout(LayoutKind.Sequential, Size = 32)]
+    public struct Vector256<T> where T : struct 
+    {
+        public static bool IsSupported {get;}
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs]: The IsSupported properties should return false. (And the intrinsic JIT implementation should override it to return true.)

File: src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs (hunk 3 above)
Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs]: Done.

File: src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs (hunk 4 above)
Comment (fiigii) [reply src/mscorlib/src/System/Runtime/Intrinsics/Vector256.cs]: Done.

**Comment by fiigii** [comment]
Change build condition to AnyCPU for leaving these intrinsics as stubs on other platforms. Successfully build with CoreFX change. @jkotas PTAL.

**Comment by fiigii** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by fiigii** [comment]
## Update
Separate SSE intrinsics from `Sse2` class. New file Sse.cs and class `Sse` added.

**Comment by fiigii** [comment]
@jkotas I have addressed all the feedback and rebased to solve conflicts. Could you please take a look?

**Comment by jkotas** [comment]
LGTM

**Comment by fiigii** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test 
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by fiigii** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 full_opt legacy_backend CoreCLR Perf Tests Correctness

**Comment by fiigii** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by fiigii** [comment]
@jkotas Could you please merge this PR? Then we can test and build the CoreFX counterpart in the CI system.

**Pull request merged**